### DISCUSSION TITLE
Component does not find Opp Parent Record

### DISCUSSION THREAD

**Discussion by ChristianK1711** [deejay-hub/timeline-lwc#179] [Q&A] [answered]
Hi Dave, 

we did a fresh install of the timeline in our fully copy sandbox. Tried it on account and asset and it nicely works. 

When I put the component on the opportunity lightning record page, it does not find the "Use This Opportunity" Parent Record. 

The custom metadata comes with a preset of related objects to be shown on opportunity and all are active, looking ok. 

![image](https://user-images.githubusercontent.com/102682152/160875645-214c457b-4de1-43e4-8f4c-225f06aab6c2.png)

I checked the installation guide and KB but was not able to figure sth. out here. Any idea? 

Thanks! 

P.S.: Our organization will love this! 

Cheers, Christian

**Comment by deejay-hub** [comment]
Hi @ChristianK1711 sounds like a strange one I've not come across before. This field tries to populate all fields on the Opportunity object where there is a relationship. If for some reason one of the fields on your Opportunity object causes an issue then I could foresee this error happening.

The only way we have to check is to run the following code in the developer console (execute apex anonymous). This is the code that loops through the object to find the fields to populate into the dropdown list. I have amended the code to output the field names to the debug log.
```

		Schema.DescribeSObjectResult describeSobjects = ((SObject) (Type.forName(
                        'Schema.' + 'Opportunity'
                    )
                    .newInstance()))
                .getSObjectType()
                .getDescribe(SObjectDescribeOptions.DEFERRED);

            Map<String, Schema.SObjectField> myFields = describeSobjects.fields.getMap();

            for (String field : myFields.keySet()) {
                Schema.DescribeFieldResult currentField = myFields.get(field).getDescribe();

                if (
                    currentField.isAccessible() &&
                    currentField.isNamePointing() == false &&
                    currentField.getLabel() != 'Master Record ID' &&
                    (String.valueOf(currentField.getReferenceTo()) != '' &&
                    String.valueOf(currentField.getReferenceTo()) != null &&
                    String.valueOf(currentField.getReferenceTo()) != '()' &&
                    String.valueOf(currentField.getReferenceTo()) != '(User)')
                ) {
                    System.Debug('@@@ ' + currentField.getLabel()) ;
                    System.Debug('@@@ ' + field) ;
                }
            }
```
Would you be willing to run it in the Org you are having issues with to see if it runs successfully? If you need step by step screenshots just let me know.

**Reply by ChristianK1711** [reply]
Dear Dave @deejay-hub , 

thank you so much for your attention and advise! 

I did run the code in the respective org and here is the log: 
[apex-07L5E00000RwkbgUAB.log](https://github.com/deejay-hub/timeline-lwc/files/8386817/apex-07L5E00000RwkbgUAB.log)

Do you see anything wrong that could lead to the solution? 

Regards & Thanks, Christian

**Reply by deejay-hub** [reply]
Nothing so far. There are 26 fields identified to be put into the drop down and I've recreated all labels for length and characters to try and replicate what you see but no joy so far - it seems to be okay in a fresh scratch org.

We'll add some additional error handling in the next release to look for unexpected exceptions and always display the 'Use This x' option even if there's an error adding specific fields from the Object.

In the meantime I will keep this open and think about it some more.

Thanks for your prompt help so far.

**Reply by ChristianK1711** [reply]
Wow, thank you very much for your investigation. 

When do you expect the release making sure the "Use this xxx" option is always shown and not disrupted? 

I'll create a new dev sandbox and try a fresh installation there. 

Keep you posted... 

Thank you! 

Regards, Christian

**Comment by ChristianK1711** [comment]
Quick Update: In an empty and fresh dev sandbox it works ok and all possible related options including "Use this Opportunity" are available. 

Could it be a data issue? 

We'll try it in our "official" Dev (dev sandbox)  & QA (full copy) Orgs soon and let you know here. 

Thanks!

**Comment by deejay-hub** [comment]
Thanks for checking @ChristianK1711 good to know it worked in a fresh sandbox. It's likely to be a metadata issue...a label...or translated label in your setup in the full sandbox that cannot be displayed..... is my best guess...but at this stage I'm not ruling anything out.

Having better error handling will help but I would love to know what causes it as there is a chance whatever code we add doesn't fix your particular issue. Next release is scheduled for 1st week of June.

**Comment by ChristianK1711** [comment]
Good morning! 

Let me check how it works when we're going into our "real" QA full copy sandbox and production system afterwards (the previous org where it did not work was a playground QA full copy - nevertheless a copy of the "real" QA.). 

Keep you posted here. 

Thanks!

**Comment by deejay-hub** [comment, marked as answer]
New v1.11 now contains additional error handling. Hopefully it addresses the issue here.
